kuksa_databroker/databroker/src/broker.rs

```diff
@@ -374,6 +540,10 @@ impl Entry {
             changed.insert(Field::ActuatorTarget);
         }
 
+        if update.allowed != self.metadata.allowed {
+            self.metadata.allowed = update.allowed;
+        }
```

Comment:
Alright testing the actual functionality now and noticed that enforcing allowed values only works the first time.

The reason is the code added here.

It is changing `metadata.allowed` to `None` when the update doesn't contain any `allowed` (i.e. the common case). All subsequent requests are then allowed regardless of value since `allowed` is now set to `None`.

Reply:
When #507 is merged, this would actually have been prevented anyway, since changing metadata isn't allowed. But it's still a bug.

So one way to fix it is just removing this code for the time being, until we add support for updating metadata (and not just allowed values).

Reply:
This is handled for `actuator_target` by wrapping it in an additional `Option<>`

```
    // Actuator target is wrapped in an additional Option<> in
    // order to be able to convey "update it to None" which would
    // mean setting it to `Some(None)`.
    pub actuator_target: Option<Option<Datapoint>>, // only for actuators
```